### PR TITLE
Generic error

### PR DESCRIPTION
**Pull request by adodge-bcg**
# Description

This PR includes the following proposed change(s):

- Expanded error state in Home to handle incorrect navigation to site.
homeError

Initial implementation of: [FLA-269](https://justice.gov.bc.ca/jira/browse/FLA-269)

See [errorHandling doc](https://github.com/bcgov/jag-file-submission/blob/master/docs/errorHandling.md) for more information on additional errors.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Tested locally, with storybook, and Jest.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/frontend/efiling-frontend/src/components/page/home/Home.js [hunk 1]
```diff
@@ -104,14 +111,26 @@ export default function Home({ page: { header, confirmationPopup } }) {
...
+      )}
+      {!showLoader && !error && !csoAccountStatus.exists && (
         <CSOAccount
           confirmationPopup={confirmationPopup}
           applicantInfo={applicantInfo}
           setCsoAccountStatus={setCsoAccountStatus}
         />
       )}
-      {!showLoader && csoAccountStatus.exists && (
+      {!showLoader && !error && csoAccountStatus.exists && (
```
Comment (alexjoybc) [inline src/frontend/efiling-frontend/src/components/page/home/Home.js:133]: is it not the same check as above ?

File: src/frontend/efiling-frontend/src/modules/errorRedirect.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export function errorRedirect(errorUrl, error) {
+  if (
+    errorUrl &&
+    error.response &&
+    error.response.status &&
+    error.response.data &&
+    error.response.data.message
+  ) {
+    return window.open(
+      `${errorUrl}?status=${error.response.status}&message=${error.response.data.message}`,
```
Comment (alexjoybc) [inline src/frontend/efiling-frontend/src/modules/errorRedirect.js:10]: Could we document that in docsify, I think we need to start adding this to the getting started ?

File: src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.js [hunk 3]
```diff
@@ -50,11 +51,7 @@ export default function CSOAccount({
         setCsoAccountStatus({ exists: true, isNew: true });
       })
       .catch(error => {
-        const errorUrl = sessionStorage.getItem("errorUrl");
-        window.open(
-          `${errorUrl}?status=${error.response.status}&message=${error.response.data.message}`,
-          "_self"
-        );
+        errorRedirect(sessionStorage.getItem("errorUrl"), error);
```
Comment (sdevalapurkar-bcgov) [inline src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.js:54]: 👚 clean

File: src/frontend/efiling-frontend/src/components/page/home/Home.js (hunk 1 above)
Comment (sdevalapurkar-bcgov) [reply src/frontend/efiling-frontend/src/components/page/home/Home.js:133]: this one is when csoAccountExists but previous is when csoAccount does not exist 😄

File: src/frontend/efiling-frontend/src/components/page/home/Home.stories.js [hunk 4]
```diff
@@ -79,7 +79,9 @@ const noAccountExistsComponent = (
   </NoAccountExistsStateData>
 );
 
-export const Loader = () => loaderComponent;
+export const Error = () => loaderComponent;
+
+export const ErrorMobile = () => loaderComponent;
```
Comment (sdevalapurkar-bcgov) [inline src/frontend/efiling-frontend/src/components/page/home/Home.stories.js:84]: I purposely removed this story because I dont think we really needed it, but now that it is error state if you think we need it then that's good. Just wondering :)

File: src/frontend/efiling-frontend/src/modules/errorRedirect.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+      "_self"
+    );
+  }
+
+  return null;
```
Comment (sdevalapurkar-bcgov) [inline src/frontend/efiling-frontend/src/modules/errorRedirect.js:15]: do you think we can add a test case for this function testing the redirect/return of null?

File: src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.js (hunk 3 above)
Comment (sdevalapurkar-bcgov) [reply src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.js:54]: i dont even think we need to pass in this first param here, the function can be responsible for pulling it from sessionStorage

File: src/frontend/efiling-frontend/src/components/page/home/Home.js (hunk 1 above)
Comment (adodge-bcg) [reply src/frontend/efiling-frontend/src/components/page/home/Home.js:133]: One should be:
```bash
!csoAccountStatus.exists
```
and the other:
```bash
csoAccountStatus.exists
```
So if a CSO account exists, go directly to PackageConfirmation and if no CSO account exists, go to CSOAccount to create one.

File: src/frontend/efiling-frontend/src/components/page/home/Home.stories.js (hunk 4 above)
Comment (adodge-bcg) [reply src/frontend/efiling-frontend/src/components/page/home/Home.stories.js:84]: I left it in to make sure the error still looked ok on mobile...not 100% that we need it but figured I would leave/re-add it incase.

File: src/frontend/efiling-frontend/src/modules/errorRedirect.js (hunk 2 above)
Comment (adodge-bcg) [reply src/frontend/efiling-frontend/src/modules/errorRedirect.js:10]: Updated the errorHandling.md, not sure what you envision on the docsify getting started for this?

File: src/frontend/efiling-frontend/src/modules/errorRedirect.js (hunk 5 above)
Comment (adodge-bcg) [reply src/frontend/efiling-frontend/src/modules/errorRedirect.js:15]: Should be covered now.
